Library/Homebrew/formula.rb

```diff
@@ -493,8 +493,21 @@ def python(options={}, &block)
   alias_method :python3, :python
 
   # an array of all {Formula} names
-  def self.names
-    Dir["#{HOMEBREW_LIBRARY}/Formula/*.rb"].map{ |f| File.basename f, '.rb' }.sort
+  def self.names(options={:fullname => false})
+    core = Dir["#{HOMEBREW_LIBRARY}/Formula/*.rb"].map{ |f| File.basename f, ".rb" }.sort
+    taps = Dir["#{HOMEBREW_LIBRARY}/Taps/*/*/"].reduce([]) do |formulae, tap|
+      Pathname.new(tap).find_formula do |f|
+        if options[:fullname]
+          if f.to_s =~ HOMEBREW_TAP_PATH_REGEX
+            formulae << "#$1/#{$2.gsub(/^homebrew-/, "")}/#{f.basename(".rb")}"
+          end
+        else
+          formulae << f.basename(".rb").to_s
+        end
+      end
+      formulae
+    end.sort
+    (core+taps).uniq
   end
```

Comment:
Again, I'd rather this function behaved as it did before and we add a new function for returning fully-qualified names.